api_compatibility_policy.md

```diff
@@ -136,6 +136,21 @@ See the current list of [alpha features](https://github.com/tektoncd/pipeline/bl
 
 - GA/Stable features will not be removed or changed in a backwards incompatible manner without incrementing the API Version.
 
+### Feature Graduation Process
+Features are first released as experimental in alpha, refined in beta, and finalized in stable releases.
+
+#### Introducing an `alpha` feature
+- When a feature is first introduced to Tekton, it will have the `alpha` stability level and be disabled by default. 
+- At this stage, users could choose to experiment with the feature. Feedback will be collected from users and it will help maintainers to determine whether to promote the feature to higher stability level or deprecate it.
+
+#### Promoting a feature to `beta`
+- After feedback of the usage of the alpha features, once the needs and motivations are validated, a feature could be promoted to `beta`. This stage is where features are further tested and refined.
+- The dedicated feature flag for this feature will then change the stability level only. It will still be disabled by default.
```

Comment:
we don't implement per feature flag yet, what does it mean by "change the stability level only"? Updating the documentation?

Reply:
I was trying to make the point that we don't need to update separately for different apiVersions. For example, validations were different for beta features in v1 and v1beta1 before we have such indication. So we'd only need to change the accepted stability level for the validation only instead of changing separately for v1 and v1beta1.

On second thought I think we could just avoid the confusion made by "only" by rephrasing this. Please take another look for the updates:)

Reply:
Understand your point:) I've updated the wording to be more specific on the 'action' to take when promoting features.

While in this feature graduation process documentation I think it might be better to keep it more general since we are going to live with a stage where both `enable-api-fields` for **old** features and `per-feature flags` for **new** features exists. For more detailed action items for promoting features with per-feature flags, as you pointed, shall be in after the per-feature flag struct being implemented. I think they shall all go in the current v0.53.0 release.